### PR TITLE
[SPARK-15037][SQL][MLLIB] Use SparkSession instead of SQLContext in Scala/Java TestSuites

### PR DESCRIPTION
**Pull request by techaddict**
## What changes were proposed in this pull request?

Use SparkSession instead of SQLContext in Scala/Java TestSuites
as this PR already very big working Python TestSuites in a diff PR.
## How was this patch tested?

Existing tests

### CODE REVIEW

**Comment by techaddict** [comment]
cc: @andrewor14 @rxin @dongjoon-hyun PR is ready with Scala/Java Changes.

File: mllib/src/test/java/org/apache/spark/ml/classification/JavaOneVsRestSuite.java [hunk 1]
```diff
@@ -17,69 +17,72 @@
...
-
+import org.apache.spark.sql.SparkSession;
 import org.junit.After;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import scala.collection.JavaConverters;
+
+import java.io.Serializable;
+import java.util.List;
```
Comment (andrewor14) [inline mllib/src/test/java/org/apache/spark/ml/classification/JavaOneVsRestSuite.java]: import ordering is wrong, should be
```
java imports

scala imports

third-party (e.g. org, scala)

spark (org.apache.spark)
```

File: mllib/src/test/scala/org/apache/spark/ml/feature/ChiSqSelectorSuite.scala [hunk 2]
```diff
@@ -30,8 +30,8 @@ class ChiSqSelectorSuite extends SparkFunSuite with MLlibTestSparkContext
   with DefaultReadWriteTest {
 
   test("Test Chi-Square selector") {
-    val sqlContext = SQLContext.getOrCreate(sc)
-    import sqlContext.implicits._
+    val spark = SQLContext.getOrCreate(sc)
```
Comment (andrewor14) [inline mllib/src/test/scala/org/apache/spark/ml/feature/ChiSqSelectorSuite.scala]: this isn't even a `SparkSession` though

**Comment by andrewor14** [comment]
@techaddict the changes look OK. I noticed there were a few places where we didn't actually use `SparkSession` but just renamed the variable. Can you fix those?

**Comment by andrewor14** [comment]
(you also need to fix the imports in many other files)

**Comment by techaddict** [comment]
@andrewor14 test failed because `WAREHOUSE_PATH` is set to `${system:user.dir}/spark-warehouse` by default (https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L55-L58). Where as if we call warehousePath, it replaces this  with `System.getProperty("user.dir")` so it works fine (https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L654). There is definitely something wrong with the default value, not sure what to do ?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 3]
```diff
@@ -66,7 +66,8 @@ class DDLSuite extends QueryTest with SharedSQLContext with BeforeAndAfterEach {
 
   private def createDatabase(catalog: SessionCatalog, name: String): Unit = {
     catalog.createDatabase(
-      CatalogDatabase(name, "", sqlContext.conf.warehousePath, Map()), ignoreIfExists = false)
+      CatalogDatabase(name, "", spark.sessionState.conf.warehousePath, Map()),
```
Comment (andrewor14) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: it should be the same if you just do `spark.conf.warehousePath`. I actually don't understand why the tests failed earlier. Looks like it only happens to this patch.

Comment (techaddict) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: we can't do `spark.conf.warehousePath` coz conf in `SparkSession` is `RuntimeConfig` not `SQLConf`.

Comment (andrewor14) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: oh yeah

**Comment by andrewor14** [comment]
OK, this looks good. I'm going to merge this into master 2.0 first so you don't have to keep rebasing it. If there are problems we can fix them in follow-up patches later.

**Pull request closed without merging**